submarine-test/test-k8s/src/test/java/org/apache/submarine/server/client/JobManagerRestApiIT.java

```diff
@@ -0,0 +1,235 @@
+import org.apache.submarine.server.json.JobIdSerializer;
+import org.apache.submarine.server.response.JsonResponse;
+import org.apache.submarine.server.rest.RestConstants;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("rawtypes")
+public class JobManagerRestApiIT {
+  private static final Logger LOG = LoggerFactory.getLogger(JobManagerRestApiIT.class);
+
+  private static String JOB_PATH = "/api/" + RestConstants.V1 + "/" + RestConstants.JOBS;
+
+  private Gson gson = new GsonBuilder()
+      .registerTypeAdapter(JobId.class, new JobIdSerializer())
+      .registerTypeAdapter(JobId.class, new JobIdDeserializer())
+      .create();
+
+  @Test
+  public void testJobServerPing() throws IOException {
+    GetMethod response = HttpClientManager.doGet("/api/" + RestConstants.V1 + "/"
+        + RestConstants.JOBS + "/" + RestConstants.PING);
+    String requestBody = response.getResponseBodyAsString();
+    Gson gson = new Gson();
+    JsonResponse jsonResponse = gson.fromJson(requestBody, JsonResponse.class);
+    Assert.assertEquals(Response.Status.OK.getStatusCode(), jsonResponse.getCode());
+    Assert.assertEquals("Pong", jsonResponse.getResult().toString());
+  }
+
+  @Test
+  public void testTensorFlowWithJsonSpec() throws Exception {
+    String body = loadContent("tensorflow/tf-mnist-req.json");
+    String patchBody = loadContent("tensorflow/tf-mnist-patch-req.json");
+    run(body, patchBody, "application/json");
+  }
+
+  @Test
+  public void testTensorFlowWithYamlSpec() throws Exception {
+    String body = loadContent("tensorflow/tf-mnist-req.yaml");
+    String patchBody = loadContent("tensorflow/tf-mnist-patch-req.yaml");
+    run(body, patchBody, "application/yaml");
+  }
+
+  @Test
+  public void testPyTorchWithJsonSpec() throws Exception {
+    String body = loadContent("pytorch/pt-mnist-req.json");
+    String patchBody = loadContent("pytorch/pt-mnist-patch-req.json");
+    run(body, patchBody, "application/json");
+  }
+
+  @Test
+  public void testPyTorchWithYamlSpec() throws Exception {
+    String body = loadContent("pytorch/pt-mnist-req.yaml");
+    String patchBody = loadContent("pytorch/pt-mnist-patch-req.yaml");
+    run(body, patchBody, "application/yaml");
+  }
+
+  private void run(String body, String patchBody, String contentType) throws Exception {
+    // create
+    PostMethod postMethod = HttpClientManager.doPost(JOB_PATH, body, contentType);
+    Assert.assertEquals(Response.Status.OK.getStatusCode(), postMethod.getStatusCode());
+
+    String json = postMethod.getResponseBodyAsString();
+    JsonResponse jsonResponse = gson.fromJson(json, JsonResponse.class);
+    Assert.assertEquals(Response.Status.OK.getStatusCode(), jsonResponse.getCode());
+
+    Job createdJob = gson.fromJson(gson.toJson(jsonResponse.getResult()), Job.class);
+    Assert.assertEquals("mnist", createdJob.getName());
```

Comment:
We need to verify more fields in the job. Just the name and the response code is not enough.
For instance, we can verify the "uid" and the "acceptedTime" is not null because they are populated by the K8s.